### PR TITLE
WebSocket: parse channel_member_updated event

### PR DESCRIPTION
**Pull request by immindich**
The server sends this event when a user updates their preferences, so this is necessary for matterhorn-chat/matterhorn#582.

### CODE REVIEW

**Comment by immindich** [comment]
@jtdaugherty I think this should be ready to merge, pending a review. This should be enough to support channel muting. I'll be spending some time this afternoon and tomorrow working on the UI changes related to this. There may be more changes required to support updating the other notification preferences from the client, but those can go in a separate PR.

File: src/Network/Mattermost/WebSocket/Types.hs [hunk 1]
```diff
@@ -235,6 +236,10 @@ instance FromJSON WEData where
     wepPreferences <- case wepPreferencesRaw of
       Just str -> fromValueString str
       Nothing  -> return Nothing
+    wepChannelMemberRaw <- o .:? "channelMember"
+    wepChannelMember <- case wepChannelMemberRaw of
```
Comment (jtdaugherty) [inline src/Network/Mattermost/WebSocket/Types.hs]: This bind/`case` pair can be collapsed to `wepChannelMember <- fmap fromValueString <$> o .:? "channelMember"`.

**Comment by jtdaugherty** [comment]
Thanks! I took a look and left a comment but otherwise it looks good to me. I also just confirmed with the server developers that this websocket event was added in 4.9.0, so we won't need to worry about bumping the major server version components of the library version (or Matterhorn's version).

File: src/Network/Mattermost/WebSocket/Types.hs (hunk 1 above)
Comment (immindich) [reply src/Network/Mattermost/WebSocket/Types.hs]: It ended up being slightly different due to the type of `fromValueString`. I also applied the same simplification to the other bindings in that `do` block.

**Comment by immindich** [comment]
One thing I just remembered is that this also adds the `channelNotifyPropsIgnoreChannelMentions` field to `ChannelNotifyProps`. It seems like this is undocumented in the API reference.

**Comment by jtdaugherty** [comment]
Oh, it isn't documented in the server API docs? If not, that'd be great to pass on to Harrison H. in the `matterhorn` channel on `community.mattermost.com`.

**Pull request merged**